target_chains/sui/contracts/sources/governance/governance.move

```diff
@@ -7,43 +7,106 @@ module pyth::governance {
     use pyth::set_fee_recipient;
     use pyth::state::{Self, State};
     use pyth::set_update_fee;
-    use pyth::governance_witness::{GovernanceWitness};
 
-    use wormhole::governance_message::{Self, DecreeReceipt};
+    use wormhole::state::{State as WormState, Self as worm_state};
+    use wormhole::vaa::{Self, VAA};
+    use wormhole::bytes32::Bytes32;
 
     const E_INVALID_GOVERNANCE_ACTION: u64 = 0;
     const E_MUST_USE_CONTRACT_UPGRADE_MODULE_TO_DO_UPGRADES: u64 = 1;
     const E_CANNOT_EXECUTE_GOVERNANCE_ACTION_WITH_OBSOLETE_SEQUENCE_NUMBER: u64 = 2;
+    const E_OLD_GUARDIAN_SET_GOVERNANCE: u64 = 3;
+    const E_INVALID_GOVERNANCE_CHAIN: u64 = 4;
+    const E_INVALID_GOVERNANCE_EMITTER: u64 = 5;
+
+    // this struct does not have the store or key ability so it must be
+    // used in the same txn chain in which it is created
+    struct WormholeVAAVerificationReceipt{
+        payload: vector<u8>,
+        digest: Bytes32,
+        sequence: u64,  // used for replay protection
+    }
+
+    public fun take_payload(receipt: &WormholeVAAVerificationReceipt): vector<u8> {
+        receipt.payload
+    }
+
+    public fun take_digest(receipt: &WormholeVAAVerificationReceipt): Bytes32 {
+        receipt.digest
+    }
+
+    public fun destroy(receipt: WormholeVAAVerificationReceipt) {
+        let WormholeVAAVerificationReceipt{payload: _, digest: _, sequence: _} = receipt;
+    }
+
+   // We define a custom verify_vaa function instead of using wormhole::governance_message::verify_vaa
+   // because that function makes extra assumptions about the VAA payload headers. Pyth uses a
+   // different header format compared to Wormhole, so
+    public fun verify_vaa(
```

Comment:
I believe we don't need additional verification on top of what wormhole is doing in the general `parse_and_verify_vaa` function. The exact match of the guardian_set_index is only necessary for wormhole upgrades, not ours. We just need to accept a `VAA` and check the data source with `is_valid_governance_data_source`

Reply:
I think you're right about not having to check the guardian set index here, but I think we still need to check that the emitter address and emitter chain correspond to Pyth governance address and chain.

Reply:
I believe that's done in `is_valid_governance_data_source` right?